packages/arb-avm-cpp/data_storage/src/arbcore.cpp

```diff
@@ -1833,18 +1841,17 @@ std::optional<std::vector<value>> ArbCore::logsCursorGetLogs(
         return std::nullopt;
     }
 
-    logs_cursors[cursor_index].pending_total_count =
-        logs_cursors[cursor_index].current_total_count +
-        logs_cursors[cursor_index].data.size();
-
-    std::vector<value> logs{std::move(logs_cursors[cursor_index].data)};
+    std::vector<value> logs = std::move(logs_cursors[cursor_index].data);
     logs_cursors[cursor_index].data.clear();
 
-    return logs;
+    first_index_out = logs_cursors[cursor_index].current_total_count;
+
+    return std::move(logs);
```

Comment:
`std::move` on returns kills NRVO. Just use `return logs` instead here and in `logsCursorGetDeletedLogs`

Reply:
refactor made this not applicable